java/lib/build.gradle.kts

```diff
@@ -29,14 +25,31 @@ java {
 }
 
 tasks.named<Test>("test") {
-    // Use JUnit Platform for unit tests.
-    useJUnitPlatform()
-    // Set the JVM argument for the java.library.path (To import rust compiled library)
-    val rustLibPath = projectDir.resolve("../../target/release").absolutePath
+    val rustLibPath = projectDir.resolve("../../target/release").absolutePath  // Set the JVM argument for the java.library.path (To import rust compiled library)
+    useJUnitPlatform()                                                         // Use JUnit Platform for unit tests.
     jvmArgs = listOf("-Djava.library.path=$rustLibPath")
 }
 
 tasks.named<Javadoc>("javadoc") {
     options.encoding = "UTF-8"
     options.memberLevel = JavadocMemberLevel.PUBLIC
 }
+
+publishing {
+    // Define a Maven publication for the 'maven' repository
+    publications {
+        create<MavenPublication>("Maven") {
+            from(components["java"])
+        }
```

Comment:
We can also add some necessary metadata here:

```kotlin
        create<MavenPublication>("mavenJava") {
            pom {
                name = "My Library"
                description = "A concise description of my library"
                url = "http://www.example.com/library"
                properties = mapOf(
                    "myProp" to "value",
                    "prop.with.dots" to "anotherValue"
                )
                licenses {
                    license {
                        name = "The Apache License, Version 2.0"
                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
                    }
                }
                developers {
                    developer {
                        id = "johnd"
                        name = "John Doe"
                        email = "john.doe@example.com"
                    }
                }
                scm {
                    connection = "scm:git:git://example.com/my-library.git"
                    developerConnection = "scm:git:ssh://example.com/my-library.git"
                    url = "http://example.com/my-library/"
                }
            }
        }

```

Reply:
Interesting! Seems a bit overkill, but thanks for pointing that out. I'll include at least a name and description to it

Reply:
Done!